freeport.go

```diff
@@ -15,35 +15,28 @@ func GetFreePort() (int, error) {
 	if err != nil {
 		return 0, err
 	}
-	defer l.Close()
+	l.Close()
 	return l.Addr().(*net.TCPAddr).Port, nil
 }
 
-// GetPort is deprecated, use GetFreePort instead
-// Ask the kernel for a free open port that is ready to use
-func GetPort() int {
+// MustGetFreePort calls GetFreePort and panics on error
+func MustGetFreePort() int {
 	port, err := GetFreePort()
 	if err != nil {
 		panic(err)
 	}
 	return port
 }
 
-// GetFreePort asks the kernel for free open ports that are ready to use.
+// GetFreePorts asks the kernel for free open ports that are ready to use.
 func GetFreePorts(count int) ([]int, error) {
-	var ports []int
+	ports := make([]int, count)
 	for i := 0; i < count; i++ {
-		addr, err := net.ResolveTCPAddr("tcp", "localhost:0")
+		p, err := GetFreePort()
```

Comment:
Calling `GetFreePort` here instead of finding and listening on the port, then `defer`-ing the `Close` call, will make it so that `GetFreePorts` may return duplicate ports.

Currently it doesn't return duplicate ports because it stays listening on all the ports it finds until the method is finished. If you close a port before subsequent calls to `ResolveTCPAddr`, those calls may return back the port that you just closed.

Reply:
Agreed.